File: app/src/main/java/com/tymphany/sdklist/filter/ble/FilterAdapter.java
package com.tymphany.sdklist.filter.ble;

import android.bluetooth.BluetoothDevice;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class FilterAdapter extends RecyclerView.Adapter<FilterAdapter.MyViewHolder> implements View.OnClickListener, View.OnLongClickListener{
    private final List<BluetoothDevice> list;
    private final Context context;
    private OnItemClickListener onItemClickListener;

    public FilterAdapter(Context context, List<BluetoothDevice> list) {
        this.context = context;
        this.list = list;
    }

    public void removeData(int position) {
        list.remove(position);
        notifyItemRemoved(position);
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_recycler, parent, false);
        MyViewHolder holder = new MyViewHolder(view);
        view.setOnClickListener(this);
        view.setOnLongClickListener(this);
        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        holder.itemView.setTag(position);
        BluetoothDevice device = list.get(position);
        holder.info.setText(device.getName());
        holder.rssi.setText(device.getAddress());
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public interface OnItemClickListener {
        void onItemClick(View view, int position);
        void onItemLongClick(View view, int position);
    }

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }

    @Override
    public void onClick(View view) {
        if (this.onItemClickListener != null) {
            this.onItemClickListener.onItemClick(view, (int)view.getTag());
        }
    }

    @Override
    public boolean onLongClick(View view) {
        if (this.onItemClickListener !=  null) {
            this.onItemClickListener.onItemLongClick(view, (int)view.getTag());
        }
        return true;
    }

    class MyViewHolder extends RecyclerView.ViewHolder {
        TextView info;
        TextView rssi;
        public MyViewHolder(View view) {
            super(view);
            info = (TextView)view.findViewById(R.id.info);
            rssi = (TextView)view.findViewById(R.id.rssi);
        }
    }
}

File: app_kotlin/src/main/java/com/tymphany/sdklist/filter/app_kotlin/FilterAdapter.kt
package com.tymphany.sdklist.filter.app_kotlin

import android.bluetooth.BluetoothDevice
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.View.OnLongClickListener
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class FilterAdapter(private val context: Context, private val list: MutableList<BluetoothDevice?>?) : RecyclerView.Adapter<FilterAdapter.MyViewHolder>(), View.OnClickListener, OnLongClickListener {
    private var onItemClickListener: OnItemClickListener? = null
    fun removeData(position: Int) {
        list?.removeAt(position)
        notifyItemRemoved(position)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view: View = LayoutInflater.from(context).inflate(R.layout.item_recycler, parent, false)
        val holder: MyViewHolder = MyViewHolder(view)
        view.setOnClickListener(this)
        view.setOnLongClickListener(this)
        return holder
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.itemView.tag = position
        val device = list?.get(position)
        holder.info.text = device?.name
        holder.rssi.text = device?.address
    }

    override fun getItemCount(): Int {
        return list?.size!!
    }

    interface OnItemClickListener {
        fun onItemClick(view: View?, position: Int)
        fun onItemLongClick(view: View?, position: Int)
    }

    fun setOnItemClickListener(onItemClickListener: OnItemClickListener?) {
        this.onItemClickListener = onItemClickListener
    }

    override fun onClick(view: View) {
        if (onItemClickListener != null) {
            onItemClickListener!!.onItemClick(view, view.tag as Int)
        }
    }

    override fun onLongClick(view: View): Boolean {
        if (onItemClickListener != null) {
            onItemClickListener!!.onItemLongClick(view, view.tag as Int)
        }
        return true
    }

    inner class MyViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var info: TextView = view.findViewById<View>(R.id.info) as TextView
        var rssi: TextView = view.findViewById<View>(R.id.rssi) as TextView
    }
}